crate/operator/prometheus.py

```diff
@@ -13,14 +33,51 @@
     }
 )
 
-cluster_status_gauge = Gauge(
-    "cloud_clusters_health",
-    documentation="0->GREEN, 1->YELLOW, 2->RED, 3->UNREACHABLE",
-    labelnames=["cluster_id"],
-)
+CLUSTER_METRICS = {}
```

Comment:
tldr; version of the change - we keep the metrics in-memory in this dictionary, which is updated by the `report_cluster_status` function, which is in turn called by the `handle_ping_cratedb_status.py ` handler. If a cluster is deleted, the handler will no longer run and the metric for this cluster will timeout (`if now - metrics["last_reported"] < LAST_SEEN_THRESHOLD:`) and no longer be returned.

If anyone has a better idea, I'd love to hear it!

Reply:
sounds good. I don't think I have a better idea :)